### DISCUSSION TITLE
Did anyone manage to make it run on Colab?

### DISCUSSION THREAD

**Discussion by AlguienAlli** [declare-lab/tango#10] [General]
Greetings, I've been trying for some days now to make tango run in colab, but I always get random errors related to either Dependencies or bad import (Tango not found or location doesn't exit); Interestingly "import tango" alone do work at times, but then I cannot access the Tango class, or even a simple "print()" during script initialization doesn't show.

Here is one of the dirty codes I've tried to initialize, some lines are commented as previous attempts to fix issues. (snippet is incomplete, I need first to solve this step)

```
%cd '/content'

!git clone https://github.com/declare-lab/tango/
%cd tango
!pip install -r requirements.txt
%cd diffusers
!pip install -e .
%cd '/content/tango'

#Dependency fix (If I do not enable these lines, I get more errors down the line)
#!pip install ipython==7.34.0
#!pip install pandas==1.5.3

import IPython
import soundfile as sf
#import tango
#Below line fails as "location doesn't exist". 
from tango import Tango
#tango.__file__ always return None
print(tango.__file__)

ja = Tango("declare-lab/tango")
```

I'm getting these dependency problems along the way, I guess the versions come from diffusers setup? tango setup has different versions for some modules iirc.
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchtext 0.15.1 requires torch==2.0.0, but you have torch 1.13.1 which is incompatible.
torchdata 0.6.0 requires torch==2.0.0, but you have torch 1.13.1 which is incompatible.
google-colab 1.0.0 requires ipython~=7.34.0, but you have ipython 8.12.0 which is incompatible.
google-colab 1.0.0 requires pandas~=1.5.3, but you have pandas 1.4.1 which is incompatible.

Many random errors after this depending on what commented lines I change, etc
```

Overall I think I'm missing something important, a rookie mistake maybe, but I can't find anywhere anyone with a colab notebook. Please note I'm not trying to setup UI similar to HuggingFace Spaces, I just need the command line version to work.

Thanks!

**Comment by deepanwayx** [comment]
Can you share the colab notebook? I can take a look into it.

**Reply by AlguienAlli** [reply]
Hey thanks for answering! Well is not much different than what I posted, but here is a link https://colab.research.google.com/drive/1AffxnlWiSqxkfTHAo66584H6U9wdZm0G?usp=sharing

**Comment by deepanwayx** [comment]
Managed to get it up and running here: https://colab.research.google.com/drive/1zwmomOqaUwLtQVjLQUcEEF5USB9aaRHY?usp=sharing

You need a couple of tricks to make this work:

1. Start the notebook and execute only the first cell: this will install the requirements.
2. Restart the runtime and start execution from the second cell: this will import the requirements and then you can generate audios with tango.

**Reply by AlguienAlli** [reply]
Quite a piece of work colab can be at times XD. 

Thank you very much! Just did a quick run and it worked fine.
